### PR TITLE
adds support for filter function

### PR DESCRIPTION
**Pull request by jonschlinkert**
to provide control over eliminating files like `.DS_Store` on mac or `Thumbs.db` on windows from causing an unexpected result. Also:
- adds `exists` function, since directories that do not exist should return `false`.
- updated metadata in package.json per latest npm recommendations and conventions
- unit tests for filter function
- docs for filter function

If this is too much for this lib, no worries. If it's okay, that would be great, I'd love to use this in `verb` and `generate`. thanks!

### CODE REVIEW

**Comment by jonschlinkert** [comment]
oh, one more thing... I used the `try-open` lib for the `exists` check, since `fs.exists` is deprecated

**Comment by tkellen** [comment]
hey @jonschlinkert, this looks nice! one question for ya though... why the preference for excluding `.DS_Store` automatically but not `Thumbs.db`?

**Comment by jonschlinkert** [comment]
Just my best guess at correct behavior, mainly since `.DS_Store` is created by the OS automatically after a folder is created, but `Thumbs.db` will only be there after images have been in the folder at some point.

**Comment by tkellen** [comment]
Gotcha. Would you be willing to remove the `.DS_Store` addition for OSX? It feels like an overreach. I'm :+1: to including a sample for that in the docs though!

**Comment by tkellen** [comment]
PS: I've just added you as a collaborator on the repo and an author on the package.

**Comment by jonschlinkert** [comment]
> It feels like an overreach

yeah no problem. It's it bit heavy-handed in retrospect. I'll update the pr. 

> I've just added you as a collaborator on the repo and an author on the package.

thanks!

**Comment by jonschlinkert** [comment]
I removed the `.DS_Store` code. I didn't even think about that until you said something, which is strange since I have a tendency to try to over-generalize lol

**Comment by tkellen** [comment]
Hah. I was thinking the same thing--this doesn't seem like something @jonschlinkert would add!

File: package.json [hunk 1]
```diff
@@ -2,34 +2,27 @@
...
-    "test": "mocha -R spec test"
+    "test": "mocha"
+  },
+  "dependencies": {
+    "try-open": "^0.1.0"
   },
   "devDependencies": {
-    "chai": "~1.9.1",
-    "mocha": "~1.18.2"
+    "chai": "*",
```
Comment (tkellen) [inline package.json]: any reason in particular you've unlocked the deps for mocha/chai?

Comment (jonschlinkert) [inline package.json]: bah, sorry I ran an "updater" by accident on package.json, I meant to revert that (and thought I did, so I must have run it twice! fwiw, ~~the reason the `updater` removed `mocha` is b/c it's not `require`d anywhere in the project~~. The `chai` wildcard is a hard-coded default I added a long time ago for a reason I can no longer remember.

anyway, I have two strong habits that are hard _not to do_ before I push up: run `update` and run `deps` (for lint-deps). but I try to only do this on my own projects though. so again, apologies.

Comment (jonschlinkert) [inline package.json]: also I can push up a fix for the deps in a bit. sorry again

Comment (tkellen) [inline package.json]: no worries

Comment (tkellen) [inline package.json]: can you give me a ping when ready?

**Comment by jonschlinkert** [comment]
absolutely, won't take more than a minute - just finishing up an obligation first. It will probably be tomorrow. thanks for pinging me!

File: package.json (hunk 1 above)
Comment (jonschlinkert) [inline package.json]: done. please let me know if I missed something. feel free to squash

**Comment by tkellen** [comment]
published!

**Comment by jonschlinkert** [comment]
thanks!

**Comment by tkellen** [comment]
thank you!

**Pull request closed without merging**